### PR TITLE
clean up `tree::{map,set}`

### PR DESCRIPTION
**Pull request by apasel422**

### CODE REVIEW

File: src/tree/map.rs [hunk 1]
```diff
@@ -885,7 +884,7 @@ macro_rules! define_iterator {
         item!(impl<'a, K, V> $name<'a, K, V> {
             #[inline(always)]
             fn next_(&mut self, forward: bool) -> Option<(&'a K, &'a $($addr_mut)* V)> {
-                while !self.stack.is_empty() || !self.node.is_null() {
```
Comment (Gankro) [inline src/tree/map.rs]: Did you bench this change?

File: src/tree/map.rs [hunk 2]
```diff
@@ -897,8 +896,7 @@ macro_rules! define_iterator {
                             self.node = $deref(next_node);
                         }
                         self.stack.push(node);
-                    } else {
-                        let node = self.stack.pop().unwrap();
+                    } else if let Some(node) = self.stack.pop() {
```
Comment (Gankro) [inline src/tree/map.rs]: Couldn't this just be an `else { return self.stack.pop().map(...) }`?

Comment (apasel422) [inline src/tree/map.rs]: I believe so, yes. Good point.

File: src/tree/map.rs (hunk 1 above)
Comment (apasel422) [inline src/tree/map.rs]: No, though I would expect it to be faster since each condition is only checked once. Let me get some metrics.

**Comment by Gankro** [comment]
@apasel422 for FOTT :heart:

File: src/tree/map.rs (hunk 1 above)
Comment (apasel422) [inline src/tree/map.rs]: Before:
```
test tree::map::bench::iter_1000                                  ... bench:      4675 ns/iter (+/- 549)
test tree::map::bench::iter_100000                                ... bench:   1245493 ns/iter (+/- 253063)
test tree::map::bench::iter_20                                    ... bench:        66 ns/iter (+/- 9)
```
After:
```
test tree::map::bench::iter_1000                                  ... bench:      3098 ns/iter (+/- 438)
test tree::map::bench::iter_100000                                ... bench:   1084854 ns/iter (+/- 219989)
test tree::map::bench::iter_20                                    ... bench:        65 ns/iter (+/- 2)
```

Comment (Gankro) [inline src/tree/map.rs]: Nice!

**Comment by Gankro** [comment]
Thanks a bunch!

**Pull request merged**